### DISCUSSION TITLE
Expected behavior for Swagger enum classes in request body

### DISCUSSION THREAD

**Discussion by smccamy1** [fastapi/fastapi#7594] [Questions] [answered]
What is the expected behavior for Swagger when you have an enum class of options for a request body?  The auto generated example is showing the first element ("block") but not sure if there should be a list of options there?  The schema shows an Emum Array with the elements listed inside a single array.  Validation is working.  My goal is to show the options within the docs if that is possible.  Just wondering if I am missing something here.  

```
class Actions(str, Enum):
    block = "block"
    unblock = "unblock"

class Site(BaseModel)
    actions: Actions
```

I have read this but I didn't see an example so I am not sure what the expected behavior is.
https://github.com/tiangolo/fastapi/issues/329

**Comment by phy25** [comment, marked as answer]
> The auto generated example is showing the first element ("block") but not sure if there should be a list of options there?

Since it's a JSON request body, and an example, I don't think it should ever render a dropdown. It should just render an example value if Swagger does not have the ability to generate the large JSON on the fly for you. The dropdown you see in #329 is in a "parameter" tab that you can change the demo request you want to send, when you "Try it out", rather than just an example.

Anyway there is definitely nothing to do with fastapi here, you may want to discuss this at https://github.com/swagger-api/swagger-ui/.

**Comment by smccamy1** [comment]
Ok, that makes sense and thanks.  The Swagger Editor shows several examples of this as well which has proven useful and figured I would mention for anyone that may stumble across this.
https://editor.swagger.io/

**Comment by tiangolo** [comment]
Thanks for the help here @phy25 ! :clap: :bow: 

Thanks for reporting back and closing the issue @smccamy1  :+1:
